### ISSUE TITLE
How to load local self-trained lora weights in the inference script?

### ISSUE THREAD

**Issue by BryantGary** [CSU-JPG/Glance#8]

**Comment by zhuobaidong** [comment]
Like this: 
pipeline.load_lora_weights("/path/to/your/trained/lora/pytorch_lora_weights.safetensors")

**Comment by BryantGary** [comment]
Thank you for your response. I also have another question: I used the multi-gpu inference script to test the effect of lora. When I used the prompt: "This image depicts a unique natural landscape featuring a winding spiral valley. The valley is characterized by its multi-layered spiral structure, with each level covered in lush green vegetation, giving it a vibrant and lively appearance. Spanning across the valleys are several ancient stone bridges, constructed from sturdy stones, exuding a sense of antiquity and solidity. In the background, rolling hills and an expansive sky can be seen, contributing to the overall serene and harmonious atmosphere of the scene." , I found that when I not use lora by  deleting the code of "load_lora_weights", the base model can still generate great results in few(5*2) steps. When I use self-trained loras, I found the generated results almost the same as not injecting loras. Why is that? 

> Like this: pipeline.load_lora_weights("/path/to/your/trained/lora/pytorch_lora_weights.safetensors")

**Comment by zhuobaidong** [comment]
Can I see the results like generated images? Sometimes for some easy prompts base model could generate great results in fewer steps, but it was not very common

**Comment by BryantGary** [comment]
<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/28339dae-1662-4c66-913d-fe79f1d4d721" />
<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/f293033d-8e29-4aac-82ed-8ccfe2287e16" />
The first is not using lora，the second is using self-trained loras，based on the aforementioned prompt and original settings in the multi-gpu infer py

**Comment by BryantGary** [comment]
> Can I see the results like generated images? Sometimes for some easy prompts base model could generate great results in fewer steps, but it was not very common

By the way，when I try to load loras, it warns like this： /opt/conda/lib/python3.10/site-packages/peft/tuners/tuners_utils.py:630: RuntimeWarning: While injecting the PEFT adapters, an inconsistency was discovered between the PEFT config and the provided state_dict. This is not necessarily an issue and can be ignored if this was the intent. The PEFT config contained these additional target modules: ['transformer_blocks.0.attn.to_k', 'transformer_blocks.0.attn.to_out.0', 'transformer_blocks.0.attn.to_q', 'transformer_blocks.0.attn.to_v', 'transformer_blocks.1.attn.to_k', 'transformer_blocks.1.attn.to_out.0', 'transformer_blocks.1.attn.to_q', 'transformer_blocks.1.attn.to_v', 'transformer_blocks.10.attn.to_k', 'transformer_blocks.10.attn.to_out.0', 'transformer_blocks.10.attn.to_q', 'transformer_blocks.10.attn.to_v', 'transformer_blocks.11.attn.to_k', 'transformer_blocks.11.attn.to_out.0', 'transformer_blocks.11.attn.to_q', 'transformer_blocks.11.attn.to_v', 'transformer_blocks.12.attn.to_k', 'transformer_blocks.12.attn.to_out.0', 'transformer_blocks.12.attn.to_q', 't.attn.to_k', 'transformer_blocks.32.attn.to_out.0', 'transformer_blocks.32.attn.to_q', 'transformer_blocks.32.attn.to_v', 'transformer_blocks.33.attn.to_k', 'transformer_blocks.33.attn.to_out.0', 'transformer_blocks.33.attn.to_q', 'transformer_blocks.33.attn.to_v', 'transformer_blocks.34.attn.to_k', 'transformer_blocks.34.attn.to_out.0', 'transformer_blocks.34.attn.to_q', 'transformer_blocks.34.attn.to_v', 'transformer_blocks.35.attn.to_k', 'transformer_blocks.35.attn.to_out.0', 'transformer_blocks.35.attn.to_q', 'transformer_blocks.35.attn.to_v', 'transformer_blocks.36.attn.to_k', 'transformer_blocks.36.attn.to_out.0', 'transformer_blocks.36.attn.to_q', 'transformer_blocks.36.attn.to_v', 'transformer_blocks.37.attn.to_k', 'transformer_blocks.37.attn.to_out.0', 'transformer_blocks.37.attn.to_q', 'transformer_blocks.37.attn.to_v', 'transformer_blocks.38.attn.to_k', 'transformer_blocks.38.attn.to_out.0', 'transformer_blocks.38.attn.to_q', 'transformer_blocks.38.attn.to_v', 'transformer_blocks.39.attn.to_k', 'tr.58.attn.to_q', 'transformer_blocks.58.attn.to_v', 'transformer_blocks.59.attn.to_k', 'transformer_blocks.59.attn.to_out.0', 'transformer_blocks.59.attn.to_q', 'transformer_blocks.59.attn.to_v', 'transformer_blocks.6.attn.to_k', 'transformer_blocks.6.attn.to_out.0', 'transformer_blocks.6.attn.to_q', 'transformer_blocks.6.attn.to_v', 'transformer_blocks.7.attn.to_k', 'transformer_blocks.7.attn.to_out.0', 'transformer_blocks.7.attn.to_q', 'transformer_blocks.7.attn.to_v', 'transformer_blocks.8.attn.to_k', 'transformer_blocks.8.attn.to_out.0', 'transformer_blocks.8.attn.to_q', 'transformer_blocks.8.attn.to_v', 'transformer_blocks.9.attn.to_k', 'transformer_blocks.9.attn.to_out.0', 'transformer_blocks.9.attn.to_q', 'transformer_blocks.9.attn.to_v']. 
  warnings.warn(warning_msg, RuntimeWarning)。 Does it affect the results？

**Comment by BryantGary** [comment]
> <img alt="Image" width="1024" height="1024" src="https://private-user-images.githubusercontent.com/41322427/530916178-28339dae-1662-4c66-913d-fe79f1d4d721.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjcwODEwNDEsIm5iZiI6MTc2NzA4MDc0MSwicGF0aCI6Ii80MTMyMjQyNy81MzA5MTYxNzgtMjgzMzlkYWUtMTY2Mi00YzY2LTkxM2QtZmU3OWYxZDRkNzIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjMwVDA3NDU0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2ZjcyMDg4YmRkNDgyMWNkNjI4MDY2ZGU1OWUwZGFlN2NiZGY5NjVmMGNkYjkwZGUxODZjZjc2MzBmOTVlNmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.LgxTfG67pwJ4OiH10aIr7Wy0vrdKcWJ3tuBUqRMrpiU"> <img alt="Image" width="1024" height="1024" src="https://private-user-images.githubusercontent.com/41322427/530916177-f293033d-8e29-4aac-82ed-8ccfe2287e16.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjcwODEwNDEsIm5iZiI6MTc2NzA4MDc0MSwicGF0aCI6Ii80MTMyMjQyNy81MzA5MTYxNzctZjI5MzAzM2QtOGUyOS00YWFjLTgyZWQtOGNjZmUyMjg3ZTE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjMwVDA3NDU0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhYzcxOGRmMWM0MTZkZTAzZjZiOTRkN2M5ZGZlODFmMzgwOWEwOTYyYWMyMzQ2ZWY0YjFhOGRjYmNjNTQ5NjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Bz2B3Jr6kJbn4dV8tvb3xdVbiKQkvg7fm9Q9lXdWPR8"> The first is not using lora，the second is using self-trained loras，based on the aforementioned prompt and original settings in the multi-gpu infer py

I am confused that why not using any loras， still generate very clear images in 5*2 inference steps.

**Comment by zhuobaidong** [comment]
There are two reasons: 
1、early timesteps matters. If you just use the scheduler’s default 10-step inference, the results are usually pretty bad, mainly because the timesteps are sampled uniformly. With our slow–fast timestep design, the timestep selection itself is already optimized for low-step inference.

2、Qwen-Image is trained and inferred with the FlowMatchEulerDiscreteScheduler. Since this scheduler uses discrete timesteps, it inherently supports jump-step sampling. That’s why its few-step inference performance still looks reasonably good.

**Comment by BryantGary** [comment]
> There are two reasons: 1、early timesteps matters. If you just use the scheduler’s default 10-step inference, the results are usually pretty bad, mainly because the timesteps are sampled uniformly. With our slow–fast timestep design, the timestep selection itself is already optimized for low-step inference.
> 
> 2、Qwen-Image is trained and inferred with the FlowMatchEulerDiscreteScheduler. Since this scheduler uses discrete timesteps, it inherently supports jump-step sampling. That’s why its few-step inference performance still looks reasonably good.

Thank you for your explanation！As you mentioned, the slow–fast timestep design actually helps generate great results a lot ，so how can I validate the improvement of my self-trained lora, since the improvement of lora seems pretty tiny when inputing easy prompt (e.g., the aforementioned prompt) .

**Comment by zhuobaidong** [comment]
Your trained lora might not be the best, cause it is effected by the training data you choose, more detais in paper ablation study.
